test/test_pumactl.rb

```diff
@@ -29,16 +29,54 @@ def test_config_file
     assert_equal "t3-pid", control_cli.instance_variable_get("@pidfile")
   end
 
-  def test_environment
-    ENV.delete 'RACK_ENV' # remove from travis
+  def test_rack_env_without_environment
+    ENV.update("RACK_ENV" => "test")
     control_cli = Puma::ControlCLI.new ["halt"]
-    assert_equal "development", control_cli.instance_variable_get("@environment")
+    assert_equal "test", control_cli.instance_variable_get("@environment")
+  end
+
+  def test_environment_without_rack_env
+    ENV.delete "RACK_ENV" # remove from travis
+    control_cli = Puma::ControlCLI.new ["halt"]
+    assert_nil control_cli.instance_variable_get("@environment")
+    control_cli = Puma::ControlCLI.new ["-e", "test", "halt"]
+    assert_equal "test", control_cli.instance_variable_get("@environment")
+  end
+
+  def test_environment_with_rack_env
+    ENV.update("RACK_ENV" => "production")
+    control_cli = Puma::ControlCLI.new ["halt"]
+    assert_equal "production", control_cli.instance_variable_get("@environment")
     control_cli = Puma::ControlCLI.new ["-e", "test", "halt"]
     assert_equal "test", control_cli.instance_variable_get("@environment")
   end
 
-  def test_config_file_exist
-    ENV.delete 'RACK_ENV' # remove from travis
+  def test_environment_specific_config_file_exist
+    ENV.delete "RACK_ENV"
+    port = 6002
+    Dir.mktmpdir do |tmp_dir|
```

Comment:
I think some of this file manipulation makes the test a bit more difficult to read. What do you think about extracting a helper like:

```rb
def with_config_file(path_to_config, port)
  path = Pathname.new(path_to_config)
  Dir.mktmpdir do |tmp_dir|
    Dir.chdir(tmp_dir) do
      FileUtils.mkdir(path.dir)
      File.open(path, "w") { |f| f << "port #{port}" }
      yield
    end
  end
end

def test_environment_specific_config_file_exist
  port = 6002
  path_to_config = "config/puma.rb"

  with_config_file(path_to_config, port) do
    control_cli = Puma::ControlCLI.new ["-e", "production", "halt"]
    assert_equal path_to_config,
      control_cli.instance_variable_get("@config_file")
  end
end
```